### PR TITLE
Logging Enhancements 

### PR DESCRIPTION
**Pull request by mvacaporale**
Enables saving of numpy arrays and pandas dataframes. 
Enables saving of any seaborn plot.

As well, this PR fixes the pruning logic. (Before, no pruning was happening for SET like models). Specifically, 
- Coactivations are tracked when module.hebbian_prune is not None:
- Modules are pruned in DSNNWeightedMag according to module.hebbian_prune
- Modules are pruned in DSNNMixedHeb according to module.hebbian_prune and module.weight_prune

### CODE REVIEW

**Comment by mvacaporale** [comment]
@lucasosouza Per the fixes to `DSNN` discussed this afternoon, I've added them to this pre-existing PR.
To recap:
- Coactivations are tracked when `module.hebbian_prune` is not None:
- Modules are pruned in `DSNNWeightedMag` according to `module.hebbian_prune`
- Modules are pruned in `DSNNMixedHeb` according to `module.hebbian_prune` and `module.weight_prune`

File: nupic/research/frameworks/dynamic_sparse/models/dsnn.py [hunk 1]
```diff
@@ -259,18 +259,18 @@ def prune(self, module):
             weight = module.m.weight.clone().detach()
             num_params = module.num_params
             corr = module.get_coactivations()
-            weight_prune_perc = module.weight_prune
+            hebbian_prune_perc = module.hebbian_prune
```
Comment (lucasosouza) [inline nupic/research/frameworks/dynamic_sparse/models/dsnn.py]: It is important to mention these changes in the pull request, the title and text only talks about logging

File: nupic/research/frameworks/dynamic_sparse/common/browser.py [hunk 2]
```diff
@@ -163,7 +222,17 @@ def _get_value(
 
         # add all data for one metric - required to plot
         for m in full_metrics:
-            stats[m + "_all"].append(progress[e][m])
+
```
Comment (lucasosouza) [inline nupic/research/frameworks/dynamic_sparse/common/browser.py]: Can't really tell what is this for - I trust you've tested the browser class after the changes, and it is working same as usual?

File: nupic/research/frameworks/dynamic_sparse/models/dsnn.py (hunk 1 above)
Comment (mvacaporale) [reply nupic/research/frameworks/dynamic_sparse/models/dsnn.py]: Good point. There's a comment above but I'll add it to the description.

Comment (mvacaporale) [reply nupic/research/frameworks/dynamic_sparse/models/dsnn.py]: Fixed

File: nupic/research/frameworks/dynamic_sparse/common/browser.py (hunk 2 above)
Comment (mvacaporale) [reply nupic/research/frameworks/dynamic_sparse/common/browser.py]: I've only made changes that add functionality and my usage locally confirms that nothing was broken in the process. 

In this logic, I simply wanted a way to add values to `stats` that didn't need any preprocessing - that is, their raw values.

**Pull request merged**